From d23ce9baabaabff17a4bf6050462edd11feb7cef Mon Sep 17 00:00:00 2001
From: Scott Murray <scott.murray@konsulko.com>
Date: Wed, 19 Apr 2023 15:55:01 -0400
Subject: [PATCH] dbc2val: add installation mechanism

Add setup.py and setup.cfg to allow installing the dbcfeederlib
module and dbcfeeder.py in a way suitable for packaging.

Upstream-Status: pending

Signed-off-by: Scott Murray <scott.murray@konsulko.com>


---
 dbc2val/setup.cfg | 30 ++++++++++++++++++++++++++++++
 dbc2val/setup.py  | 14 ++++++++++++++
 2 files changed, 44 insertions(+)
 create mode 100644 dbc2val/setup.cfg
 create mode 100644 dbc2val/setup.py

diff --git a/dbc2val/setup.cfg b/dbc2val/setup.cfg
new file mode 100644
index 0000000..03dc6d6
--- /dev/null
+++ b/dbc2val/setup.cfg
@@ -0,0 +1,30 @@
+[metadata]
+name = dbcfeeder
+author = Sebastian Schildt, Naresh Nayak, Wenwen Chen
+author_email = sebastian.schildt@de.bosch.com, naresh.nayak@de.bosch.com, wenwen.chen@de.bosch.com
+description = kuksa.val DBC feeder
+long_description = file:README.md
+long_description_content_type = text/markdown
+url=https://github.com/eclipse/kuksa.val.feeders
+project_urls=
+  Source=https://github.com/eclipse/kuksa.val.feeders/tree/master/dbc2val
+  Bug Tracker=https://github.com/eclipse/kuksa.val.feeders/issues
+classifiers =
+    Intended Audience :: Developers
+    Development Status :: 3 - Alpha
+    Environment :: Console
+    Programming Language :: Python :: 3
+    License :: OSI Approved :: Apache Software License
+    Operating System :: OS Independent
+    Topic :: Software Development
+
+license_file = ../LICENSE.txt
+
+[options]
+python_requires = >=3.6
+install_requires=
+    pyserial
+    pyyaml
+    kuksa-client
+packages=dbcfeederlib
+scripts=dbcfeeder.py
diff --git a/dbc2val/setup.py b/dbc2val/setup.py
new file mode 100644
index 0000000..65f9157
--- /dev/null
+++ b/dbc2val/setup.py
@@ -0,0 +1,14 @@
+from setuptools import setup
+
+setup(
+    version_config={
+        "template": "{tag}",
+        "dev_template": "{tag}-{ccount}",
+        "dirty_template": "{tag}-{ccount}-dirty",
+        "starting_version": "0.1.0",
+        "version_callback": None,
+        "version_file": None,
+        "count_commits_from_version_file": False
+    },
+    setup_requires=['setuptools-git-versioning'],
+)
